From 9b089546bc678e440b337e9462fd784a112ed378 Mon Sep 17 00:00:00 2001
From: Adhemerval Zanella <adhemerval.zanella@linaro.org>
Date: Thu, 24 Mar 2022 15:14:35 -0300
Subject: [PATCH 121/163] math: Fix test-totalorderl-ldbl-96 exponent setting

Clang issues the following warning:

  ../sysdeps/ieee754/ldbl-96/test-totalorderl-ldbl-96.c:49:2: error:
  implicit truncation from 'int' to bit-field changes value from 65535 to
  -1 [-Werror,-Wbitfield-constant-conversion]
          SET_LDOUBLE_WORDS (ldnx, 0xffff,
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  ../sysdeps/x86_64/fpu/math_ldbl.h:56:28: note: expanded from macro
  'SET_LDOUBLE_WORDS'
    iw_u.parts.sign_exponent = (exp);                             \
                             ^ ~~~~~
  ../sysdeps/ieee754/ldbl-96/test-totalorderl-ldbl-96.c:51:2: error:
  implicit truncation from 'int' to bit-field changes value from 65535 to
  -1 [-Werror,-Wbitfield-constant-conversion]
          SET_LDOUBLE_WORDS (ldny, 0xffff,
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  ../sysdeps/x86_64/fpu/math_ldbl.h:56:28: note: expanded from macro
  'SET_LDOUBLE_WORDS'
    iw_u.parts.sign_exponent = (exp);                             \
                             ^ ~~~~~
---
 sysdeps/ieee754/ldbl-96/test-totalorderl-ldbl-96.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sysdeps/ieee754/ldbl-96/test-totalorderl-ldbl-96.c b/sysdeps/ieee754/ldbl-96/test-totalorderl-ldbl-96.c
index 1928db672f..86f45d8269 100644
--- a/sysdeps/ieee754/ldbl-96/test-totalorderl-ldbl-96.c
+++ b/sysdeps/ieee754/ldbl-96/test-totalorderl-ldbl-96.c
@@ -46,9 +46,9 @@ do_test (void)
 	SET_LDOUBLE_WORDS (ldy, 0x7fff,
 			   (tests[i] >> 32) | 0x80000000,
 			   tests[i] & 0xffffffffULL);
-	SET_LDOUBLE_WORDS (ldnx, 0xffff,
+	SET_LDOUBLE_WORDS (ldnx, -1,
 			   tests[i] >> 32, tests[i] & 0xffffffffULL);
-	SET_LDOUBLE_WORDS (ldny, 0xffff,
+	SET_LDOUBLE_WORDS (ldny, -1,
 			   (tests[i] >> 32) | 0x80000000,
 			   tests[i] & 0xffffffffULL);
 	bool to1 = totalorderl (&ldx, &ldy);
-- 
2.37.2

